src/components/TasksMapView.js

```diff
@@ -191,9 +206,12 @@ class TasksMapView extends Component {
         identifier={ task['@id'] }
         key={ task['@id'] }
         coordinate={ task.address.geo }
-        pinColor={ pinColor(task) }
         flat={ true }
         ref={ this.markers.get(task['@id']) }>
+        <View style={ markerContainer }>
+          <View style={ markerBackgroundStyle(task) } />
+          <Icon type="FontAwesome" name={ markerIconName(task) } style={ markerIconStyle(task) } />
```

Comment:
To avoid repeating, it's possible to refactor the icon into a dedicated React Component, which contains the icon/colors logic. With props, we could configure if we want to show the type icon, or the state icon.

Something like this:
```jsx
<TaskIcon task={ task } /> {/* cube/arrow-down */}
<TaskIcon task={ task } type="type" /> {/* cube/arrow-down */}
<TaskIcon task={ task } type="state" /> {/* check/remove */}
```